src/patterns.rs

```diff
@@ -259,56 +259,52 @@ fn rewrite_tuple_pat(pats: &[ptr::P<ast::Pat>],
         pat_vec.insert(pos, dotdot);
     }
 
-    if pat_vec.is_empty() {
-        path_str
```

Comment:
That sounds good, I'll look into adding back the early return.

As for why the unit variant case doesn't get the parens added, I think it returns early from the `try_opt` at [pattern:83](https://github.com/rust-lang-nursery/rustfmt/blob/master/src/patterns.rs#L83), which means that it never enters [`rewrite_tuple_pat`](https://github.com/rust-lang-nursery/rustfmt/blob/master/src/patterns.rs#L235), and instead does formatting in [`rewrite_path`](https://github.com/rust-lang-nursery/rustfmt/blob/master/src/types.rs#L37).